src/WebJobs.Script/OutOfProc/Http/DefaultHttpWorkerService.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Formatting;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.WebApiCompatShim;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Microsoft.Azure.WebJobs.Script.Extensions;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Script.OutOfProc.Http
+{
+    public class DefaultHttpWorkerService : IHttpWorkerService
+    {
+        private readonly HttpClient _httpClient;
+        private readonly HttpWorkerOptions _httpWorkerOptions;
+        private readonly ILogger _logger;
+
+        public DefaultHttpWorkerService(IOptions<HttpWorkerOptions> httpWorkerOptions, ILoggerFactory loggerFactory)
+        {
+            _httpClient = new HttpClient();
+            _httpWorkerOptions = httpWorkerOptions.Value;
+            _logger = loggerFactory.CreateLogger<DefaultHttpWorkerService>();
+        }
+
+        internal DefaultHttpWorkerService(HttpClient httpClient, IOptions<HttpWorkerOptions> httpWorkerOptions, ILoggerFactory loggerFactory)
+        {
+            _httpClient = httpClient;
+            _httpWorkerOptions = httpWorkerOptions.Value;
+            _logger = loggerFactory.CreateLogger<DefaultHttpWorkerService>();
+        }
+
+        public Task InvokeAsync(ScriptInvocationContext scriptInvocationContext)
+        {
+            if (scriptInvocationContext.FunctionMetadata.IsSimpleHttpTriggerFunction())
+            {
+                return ProcessHttpInAndOutInvocationRequest(scriptInvocationContext);
+            }
+            return ProcessDefaultInvocationRequest(scriptInvocationContext);
+        }
+
+        internal async Task ProcessHttpInAndOutInvocationRequest(ScriptInvocationContext scriptInvocationContext)
+        {
+            _logger.LogDebug("Will invoke simple httpTrigger function:{functionName} invocationId:{invocationId}", scriptInvocationContext.FunctionMetadata.Name, scriptInvocationContext.ExecutionContext.InvocationId);
```

Comment:
nit: Can you put this with a space after the `:` and single-quotes around the values? like `Will invoke simple httpTrigger function: '{functionName}' invocationId: '{invocationId}'`. I find it a lot easier to parse when scanning logs.

Reply:
Also, this will be called so often we should move this to be a `LoggerMessage.Define` pattern so this string doesn't have to be parsed each invocation. And to give it a proper EventId.

Reply:
Can i assign any EventId? or does it have to be unique? Opened follow up item to improve logging here: https://github.com/Azure/azure-functions-host/issues/5110